src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,106 @@
+package wolox.training.models;
+
+import com.sun.istack.NotNull;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import org.postgresql.shaded.com.ongres.scram.common.util.Preconditions;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+import wolox.training.exceptions.BookNotFoundInCollectionException;
+
+@Entity
+@Table(name="users")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "USER_SEQ")
+    @SequenceGenerator(name = "USER_SEQ", sequenceName = "USER_SEQ")
+    private long id;
+
+    @NotNull
+    @Column(nullable = false, unique = true)
+    private String username;
+    @NotNull
+    @Column(nullable = false)
+    private String name;
+    @NotNull
+    @Column(nullable = false)
+    private LocalDate birthdate;
+
+    @NotNull
+    @ManyToMany(cascade = {CascadeType.MERGE, CascadeType.REFRESH})
+    private List<Book> books;
+
+    public User() {
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        Preconditions.checkNotNull(username, "Username cannot be null");
+        this.username = username;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        Preconditions.checkNotNull(name, "Name cannot be null");
```

Comment:
You can use the following method:

```
import org.apache.commons.lang.StringUtils;
 StringUtils.isEmpty ()
```
It is a method that performs both a check for null and for empty.

```suggestion
        Preconditions.checkArgument(!StringUtils.isEmpty(name), "Name cannot be null or empty");
```

Reply:
Update precondition type in set